topic-operator/src/main/java/io/strimzi/operator/topic/Zk2KafkaStreams.java

```diff
@@ -18,24 +18,23 @@ public static void main(String[] args) {
         // TODO
     }
 
-    public static void move(Zk zk, Config config, Properties kafkaProperties) {
+    public static void move(Zk zk, Config config, Properties kafkaProperties) throws Exception {
         String topicsPath = config.get(Config.TOPICS_PATH);
         TopicStore zkTopicStore = new ZkTopicStore(zk, topicsPath);
 
         KafkaStreamsConfiguration configuration = new KafkaStreamsConfiguration();
-        try {
-            configuration.start(config, kafkaProperties);
-            TopicStore ksTopicStore = configuration.getTopicStore();
-
-            zk.children(topicsPath, result -> result.map(list -> {
-                list.forEach(topicName -> {
-                    Future<Topic> ft = zkTopicStore.read(new TopicName(topicName));
-                    ft.onSuccess(ksTopicStore::create);
-                });
-                return null;
-            }));
-        } finally {
-            configuration.stop();
-        }
+        configuration.start(config, kafkaProperties)
```

Comment:
I think we're going to need something more than just a tool which the user has to run in order to migrate. I think we need the TO to do the migration automatically on start up:
```
if !exists(topic) {
    if exists(`/strimzi` znode) {
        migrate using the logic you have here. 
        if migrated ok {
            delete(`/strimzi` znode)
        } else {
            delete(topic)
            exit process
        }
    }
}
```

We're also going to need some integration test for this.

Reply:
This is just a PoC, something to discuss, WIP.

Reply:
Any progress on this?

Reply:
Not yet.
Need to think of a way to test this ...